src/roswire/proxy/roslaunch/reader.py

```diff
@@ -116,7 +141,8 @@ def wrapped(self: 'LaunchFileReader',
 
 
 @attr.s(auto_attribs=True)
-class LaunchFileReader:
+class ROS1LaunchFileReader(LaunchFileReader):
```

Comment:
#408

Reply:
Because `LaunchFileReader` defines an `app_instance` property, the auto-generated constructor for this launch file reader subclass isn't clear. If we don't have it already, we should probably add an abstract class method `.for_app_instance` to the `LaunchFileReader` interface.

Reply:
For the moment, I've made a separate factory to handle this.